optuna/visualization/_contour.py

```diff
@@ -302,7 +302,7 @@ def _generate_contour_subplot(
     # If contours_coloring='heatmap' is specified, reversescale argument of go.Contour does not
     # work correctly. See https://github.com/pfnet/optuna/issues/606.
     colorscale = plotly.colors.PLOTLY_SCALES["Blues"]
-    if direction == StudyDirection.MAXIMIZE:
+    if direction == StudyDirection.MINIMIZE:
```

Comment:
How about specifying `reversescale` to be `True` instead of manually reversing colorscale?
(https://plotly.github.io/plotly.py-docs/generated/plotly.graph_objects.Heatmap.html)

In my environment, plotting contour with enabling `reversescale` shows the same figure as this code.
`reversescale` had some problem (https://github.com/optuna/optuna/pull/662) in a heatmap, but it appears to resolve it now.

@toshihikoyanase Please let us know if you have some extra information. :pray:

Reply:
Thank you for letting me know.
I don't have additional information, but I agree with @himkt . If the plotly's bug mentioned at L301 has been resolved, we can simply revert the change in #662.

Reply:
Sounds fair if `reversescale` is working. 

Should we by the way ignore the direction completely and just fix the scale if `target` is specified and we're plotting based on something that's not necessarily the objective value? That'd be consistent with the parallel coordinate plot (but inverted) https://github.com/optuna/optuna/blob/master/optuna/visualization/_parallel_coordinate.py.

Reply:
@0x41head Could you use the `reversescale` argument here and remove the TODO comment in line 301? I think we just need to revert https://github.com/optuna/optuna/pull/662.

Reply:
Sure @HideakiImamura

Reply:
@0x41head

Thank you for the quick fix!

> Should we by the way ignore the direction completely and just fix the scale if target is specified and we're plotting based on something that's not necessarily the objective value? 

I agree. I think consistent color assigning is user-friendly.
I also feel it's reasonable to focus only on fixing color inversion in this PR.